File: rest-service/src/test/java/org/veupathdb/service/multiblast/model/blast/DustTest.java
package org.veupathdb.service.multiblast.model.blast;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

@DisplayName("Dust")
class DustTest {
  @Test
  @DisplayName("#getLevel() returns the expected value")
  void test1() {
    assertEquals(3, new Dust(3, 4, 5).getLevel());
  }

  @Test
  @DisplayName("#getWindow() returns the expected value")
  void test2() {
    assertEquals(4, new Dust(3, 4, 5).getWindow());
  }

  @Test
  @DisplayName("#getLinker() returns the expected value")
  void test3() {
    assertEquals(5, new Dust(3, 4, 5).getLinker());
  }

  @Test
  @DisplayName("#unsafeFromString() parses a correctly formatted dust string")
  void test4() {
    var target = Dust.unsafeFromString("6 7 8");

    assertEquals(6, target.getLevel());
    assertEquals(7, target.getWindow());
    assertEquals(8, target.getLinker());
  }
}

File: common/blast-types/src/test/kotlin/org/veupathdb/lib/blast/blastn/fields/DustTest.kt
package org.veupathdb.lib.blast.blastn.fields

import com.fasterxml.jackson.databind.node.ObjectNode
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.veupathdb.lib.jackson.Json


@DisplayName("Dust")
class DustTest {

  @Test
  @DisplayName("yes() returns YesDust instance.")
  fun test1() {
    assertSame(YesDust, Dust.yes())
  }

  @Test
  @DisplayName("no() returns NoDust instance.")
  fun test2() {
    assertSame(NoDust, Dust.no())
  }

  @Nested
  @DisplayName("#of(Int, Int, Int)")
  inner class Of {

    @Test
    @DisplayName("level =")
    fun test1() {
      val tgt = Dust.of(level = 10)

      assertTrue(tgt is ValDust)
      assertEquals(10, tgt.level)
      assertEquals(64, tgt.window)
      assertEquals(1, tgt.linker)
    }

    @Test
    @DisplayName("window =")
    fun test2() {
      val tgt = Dust.of(window = 10)

      assertTrue(tgt is ValDust)
      assertEquals(20, tgt.level)
      assertEquals(10, tgt.window)
      assertEquals(1, tgt.linker)
    }

    @Test
    @DisplayName("linker =")
    fun test3() {
      val tgt = Dust.of(linker = 10)

      assertTrue(tgt is ValDust)
      assertEquals(20, tgt.level)
      assertEquals(64, tgt.window)
      assertEquals(10, tgt.linker)
    }

    @Test
    @DisplayName("all")
    fun test4() {
      val tgt = Dust.of(10, 11, 12)

      assertTrue(tgt is ValDust)
      assertEquals(10, tgt.level)
      assertEquals(11, tgt.window)
      assertEquals(12, tgt.linker)
    }

  }
}
